### ISSUE TITLE
`dimaround` window rule does not work

### ISSUE THREAD

**Issue by sahinakkayadev** [hyprwm/Hyprland#1382]
**EDIT**: Check https://github.com/hyprwm/Hyprland/issues/1382#issuecomment-1382914539 to see whole config.

My decoration section is like this:

```
decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10
    blur = yes
    blur_size = 3
    blur_passes = 1
    blur_new_optimizations = on

    # active_opacity = 1
    # inactive_opacity = 0.9

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    shadow_ignore_window = true
    col.shadow = rgba(1a1a1aee)

    
    dim_inactive = true
    dim_strength = 0.1  # how much inactive windows should be dimmed, 0.0 - 1.0
    dim_around = 0.5     # how much the dimaround window rule should dim by. 0.0 - 1.0

}
```
And I have this window rule:

`windowrule=dimaround,firefox`

When I open a new firefox window, I expect to see everything around firefox to be dimmed by 0.5 but nothing happens. When I increase `dim_strength` to 0.5, I can clearly see its effects on inactive windows. But `dimaround` seems to not work at all.

**Comment by vaxerski** [comment]
dimaround is meant for floating windows, should probably be mentioned somewhere.

**Comment by sahinakkayadev** [comment]
I actually tried it with floating window too. My actual plan was to dim screen when password prompt for gpg window is opened. I tried this:


```
windowrule=float,Pinentry-gtk-2
windowrule=noborder,Pinentry-gtk-2
windowrule=dimaround,Pinentry-gtk-2
```

It still doesn't work. I am sure that the window's class name is correct because `noborder` rule is applied.

**Comment by vaxerski** [comment]
![image](https://user-images.githubusercontent.com/43317083/212494169-c0d789f5-67ca-4377-a71e-9b93ba86acc7.png)

odd. Seems to be working fine here.

**Comment by sahinakkayadev** [comment]
Can you post related bits of your config? How do you open this window? Are you setting float rule or it opens in float mode automatically?

By the way, `Pinentry-gtk-2` thing that I mention on previous comment runs on xwayland. I first thought this might be relevant but it is not because firefox still doesn't work even after setting floating rule.

**Comment by vaxerski** [comment]
![image](https://user-images.githubusercontent.com/43317083/212494499-19684284-c6a7-49eb-bf1b-07a4f8a1f7eb.png)

**Comment by sahinakkayadev** [comment]
That is really weird. I will try to debug the issue further to see if I am missing something.

**Comment by sahinakkayadev** [comment]
I checked the entire wiki and 

- It doesn't mention that `dimaround` is only available for floating windows.
- These were the only pages that mentioned dimaround: https://wiki.hyprland.org/Configuring/Variables/ and https://wiki.hyprland.org/Configuring/Window-Rules/

So there is not much left for me to do. I think I tried everything. Here is my entire config if you want to see:

```
########################################################################################
AUTOGENERATED HYPR CONFIG.
PLEASE USE THE CONFIG PROVIDED IN THE GIT REPO /examples/hypr.conf AND EDIT IT,
OR EDIT THIS ONE ACCORDING TO THE WIKI INSTRUCTIONS.
########################################################################################

#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#

# autogenerated = 1 # remove this line to remove the warning

# See https://wiki.hyprland.org/Configuring/Monitors/

monitor=,preferred,auto,1


# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 2
    gaps_out = 8
    border_size = 3
    col.active_border = rgba(1affffee) rgba(ff0000ee) 45deg
    # col.active_border = rgba(1affffee)
    col.inactive_border = rgba(595959aa)
    cursor_inactive_timeout	= 0
    layout = dwindle
    no_cursor_warps = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10
    blur = yes
    blur_size = 3
    blur_passes = 1
    blur_new_optimizations = on

    # active_opacity = 1
    # inactive_opacity = 0.9

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    shadow_ignore_window = true
    col.shadow = rgba(1a1a1aee)

    dim_around = 0.8     # how much the dimaround window rule should dim by. 0.0 - 1.0
    dimaround = 0.8     # how much the dimaround window rule should dim by. 0.0 - 1.0
    dim_inactive = false
    dim_strength = 0.2  # how much inactive windows should be dimmed, 0.0 - 1.0
}

animations {
    enabled = yes

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05
    bezier=overshot,0.05,0.9,0.1,1.1

    animation = windows, 1, 7, overshot
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}

input {
    kb_layout = us
    kb_model =
    kb_options = caps:ctrl_modifier
    kb_rules =

    numlock_by_default = true
    follow_mouse = 1

    touchpad {
        natural_scroll = yes
        drag_lock = yes
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_is_master = true
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = yes
    workspace_swipe_cancel_ratio = 0.2
}

misc {
    # If true, will animate manual window resizes/moves
    # animate_manual_resizes = false
    # Enable window swallowing
    # enable_swallow = false
    # swallow_regex = str
    # focus_on_activate = false
}

binds {
    # If enabled, an attempt to switch to the currently focused workspace will
    # instead switch to the previous workspace. Akin to i3’s auto_back_and_forth.
    workspace_back_and_forth = true
    allow_workspace_cycles = true
}

# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#executing for more
device:epic mouse V1 {
    sensitivity = -0.5
}

windowrule=float,org.telegram.desktop
windowrule=center,org.telegram.desktop
windowrule=dimaround,org.telegram.desktop

windowrule=float,Pinentry-gtk-2
windowrule=dimaround,Pinentry-gtk-2
windowrule=noborder,Pinentry-gtk-2

windowrule=float,firefox
windowrule=dimaround,firefox


$mod = SUPER
# $mod = ALT
bind = $mod, Return, exec, st
# bind = $mod, Return, exec, st -e tmux new-session -A -s st
bind = $mod, q, killactive,
bind = $mod SHIFT, q, exec, xkill
bind = $mod CONTROL, Q, exit,
binde = $mod CONTROL, n, exec, lock-screen
binde = $mod CONTROL, s, exec, toggle-screenkey
bind = $mod, e, exec, rofimoji
bind = $mod, b, exec, rofi-bluetooth
bind = $mod, v, exec, rofi-wireguard --toggle
# bind = SUPER, E, exec, [workspace 2 silent;float;noanim] st
bind = $mod, p, exec, passmenu
bind = $mod, i, exec, dictionary
bind = $mod, u, exec, smart-exec
bind = $mod, m, fullscreen, 1
bind = $mod SHIFT, m, fullscreen, 0
# bind = $mod SHIFT, m, fakefullscreen,
bind = $mod, r, swapnext,
bind = $mod, o, togglefloating,
bind = $mod, y, togglefloating,
bind = $mod, t, togglegroup, # dwindle
bind = $mod, y, pin, active
bind = $mod, semicolon, exec, wofi --show drun
bind = $mod SHIFT, O, workspaceopt, allpseudo # dwindle
bind = $mod, space, exec, hypr-toggle-layout


bind = $mod SHIFT, Return, togglesplit, # dwindle
bind = $mod, Tab, workspace, previous

# Move focus with mod + hjkl
bind = $mod, h, movefocus, l
bind = $mod, l, movefocus, r
bind = $mod, k, movefocus, u
bind = $mod, j, movefocus, d
# bind = $mod, j, changegroupactive, b
# bind = $mod, k, changegroupactive, f


# to switch between windows in a floating workspace
bind = SUPER,comma,cyclenext,          # change focus to another window
bind = SUPER,comma,bringactivetotop,   # bring it to the top
bind = SUPER,period,cyclenext, prev          # change focus to another window
bind = SUPER,period,bringactivetotop,   # bring it to the top

# resize windows with mod + control + hjkl
binde = $mod CONTROL, h, resizeactive, -25 0
binde = $mod CONTROL, l, resizeactive, 25 0
binde = $mod CONTROL, k, resizeactive, 0 -25
binde = $mod CONTROL, j, resizeactive, 0 25

# move windows with mod + shift + hjkl or mod + arrows
bind = $mod SHIFT, h, movewindow, l
bind = $mod SHIFT, l, movewindow, r
bind = $mod SHIFT, j, movewindow, d
bind = $mod SHIFT, k, movewindow, u
bind = $mod, left, movewindow, l
bind = $mod, right, movewindow, r
bind = $mod, down, movewindow, d
bind = $mod, up, movewindow, u

# Switch workspaces with mod + [0-9]
bind = $mod, a, workspace, 1
bind = $mod, s, workspace, 2
bind = $mod, d, workspace, 3
bind = $mod, f, workspace, 4
bind = $mod, g, workspace, 5
# bind = $mod, 6, workspace, 6
# bind = $mod, 7, workspace, 7
# bind = $mod, 8, workspace, 8
# bind = $mod, 9, workspace, 9
# bind = $mod, 0, workspace, 10

# Move active window to a workspace with mod + SHIFT + [0-9]
bind = $mod SHIFT, a, movetoworkspace, 1
bind = $mod SHIFT, s, movetoworkspace, 2
bind = $mod SHIFT, d, movetoworkspace, 3
bind = $mod SHIFT, f, movetoworkspace, 4
bind = $mod SHIFT, g, movetoworkspace, 5
# bind = $mod SHIFT, 6, movetoworkspace, 6
# bind = $mod SHIFT, 7, movetoworkspace, 7
# bind = $mod SHIFT, 8, movetoworkspace, 8
# bind = $mod SHIFT, 9, movetoworkspace, 9
# bind = $mod SHIFT, 0, movetoworkspace, 10

# Scroll through existing workspaces with mod + scroll
bind = $mod, mouse_down, workspace, e+1
bind = $mod, mouse_up, workspace, e-1

# Move/resize windows with mod + LMB/RMB and dragging
bindm = $mod, mouse:272, movewindow
bindm = $mod, mouse:273, resizewindow

# bind=SUPER,mouse_down,workspace,e-1
# bindl=,switch:Lid Switch,exec,lock-screen

# Example volume button that will activate even while an input inhibitor is active
bindle=, XF86AudioLowerVolume, exec, volume -5%
bindle=, XF86AudioRaiseVolume, exec, volume +5%
bindle=, XF86AudioMute, exec, volume toggle
bindle=, XF86MonBrightnessUp, exec, brightness inc 5
bindle=, XF86MonBrightnessDown, exec, brightness dec 5
bindle=, XF86KbdBrightnessUp, exec, keyboard_backlight inc
bindle=, XF86KbdBrightnessDown, exec, keyboard_backlight dec
# bindle=, Print, exec, flameshot gui
# bindr=SUPER, SUPER_L, exec, pkill wofi || wofi --show drun


# will switch to a submap called notifications
bind = $mod, n, submap, notifications

# will start a submap called "notifications"
submap=notifications

# sets repeatable binds for resizing the active window
binde=$mod, n, exec, dunstctl close
bind=$mod, n, submap, reset
binde=$mod, k, exec, dunstctl history-pop
binde=$mod, j, exec, dunstctl close
binde=$mod, m, exec, dunstctl context
bind=$mod, m, submap, reset
binde=$mod, p, exec, toggle-notifications
bind=$mod, p, submap, reset

# use reset to go back to the global submap
bind=,escape,submap,reset

# will reset the submap, meaning end the current one and return to the global one
submap=reset

# keybinds further down will be global again...

```

**Comment by sahinakkayadev** [comment]
OK, problem solved. It turns out, ```dimaround`` is not available before [v0.20](https://github.com/hyprwm/Hyprland/releases/tag/v0.20.0beta) and I was on v0.19. I was looking at my second monitor and the bar indicating the error only visible on first monitor. So I missed an obvious error, and maintainers couldn't help me because they assumed I would see the top bar indicating the error if I try to use an unavailable option. So I think you might consider implementing #1385.
